# Ultralytics Framework for YOLOv8 Training

In [1]:
import torch

# Number of GPUs available
num_gpus = torch.cuda.device_count()

print(f"Number of available GPUs: {num_gpus}")


Number of available GPUs: 4


# Yolov08n Model Training on:
- Carla_Labelling-v5 dataset
- Train: 70% = 4485 Images
- Validation: 20% = 1274 Images
- Test: 10% = 641 Images

- Hyperparameters are on automatic settings using Ultralytics package

In [2]:
from ultralytics import YOLO

# Load YOLOv10n model from scratch
model = YOLO("yolov8n.pt")

100%|██████████| 6.25M/6.25M [00:00<00:00, 99.6MB/s]


# Validate the pretrained weights by running inference on an image.

In [3]:
# Run batched inference on a list of images
results = model(["/home/sur06423/project/1.jpg"])  # return a list of Results objects

# Process results list
for result in results:
    boxes = result.boxes  # Boxes object for bounding box outputs
    masks = result.masks  # Masks object for segmentation masks outputs
    keypoints = result.keypoints  # Keypoints object for pose outputs
    probs = result.probs  # Probs object for classification outputs
    obb = result.obb  # Oriented boxes object for OBB outputs
    result.show()  # display to screen
    result.save(filename="result.jpg")  # save to disk


0: 448x640 4 horses, 55.9ms
Speed: 6.4ms preprocess, 55.9ms inference, 389.8ms postprocess per image at shape (1, 3, 448, 640)


Error: no "view" rule for type "image/png" passed its test case
       (for more information, add "--debug=1" on the command line)


# Train the model using the following guidelines:
- Link for various arguments: https://docs.ultralytics.com/modes/train/#train-settings 

In [4]:
import os

# Set library paths
os.environ['LD_LIBRARY_PATH'] = '/usr/lib/xorg-nvidia-525.116.04/lib/x86_64-linux-gnu:' + os.environ.get('LD_LIBRARY_PATH', '')
os.environ['LD_LIBRARY_PATH'] = '/usr/lib/xorg/lib/x86_64-linux-gnu:' + os.environ['LD_LIBRARY_PATH']
os.environ['LD_LIBRARY_PATH'] = '/usr/lib/xorg-nvidia-535.113.01/lib/x86_64-linux-gnu:' + os.environ['LD_LIBRARY_PATH']

# Verify the update
print(os.environ['LD_LIBRARY_PATH'])


/usr/lib/xorg-nvidia-535.113.01/lib/x86_64-linux-gnu:/usr/lib/xorg/lib/x86_64-linux-gnu:/usr/lib/xorg-nvidia-525.116.04/lib/x86_64-linux-gnu:/home/sur06423/miniconda3/envs/yolov_env/lib/python3.9/site-packages/cv2/../../lib64:


In [5]:
# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
model.train(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", # 	Path to the dataset configuration file (e.g., coco8.yaml). This file contains dataset-specific parameters, including paths to training and validation data, class names, and number of classes.
            epochs=100, # Total number of training epochs. Each epoch represents a full pass over the entire dataset. Adjusting this value can affect training duration and model performance.
            patience=10, # Number of epochs to wait without improvement in validation metrics before early stopping the training. Helps prevent overfitting by stopping training when performance plateaus.
            batch=128, # Batch size, with three modes: set as an integer (e.g., batch=16), auto mode for 60% GPU memory utilization (batch=-1), or auto mode with specified utilization fraction (batch=0.70).
            imgsz=640, # Target image size for training. All images are resized to this dimension before being fed into the model. Affects model accuracy and computational complexity.
            save=True, # Enables saving of training checkpoints and final model weights. Useful for resuming training or model deployment.
            save_period = 1, # Frequency of saving model checkpoints, specified in epochs. A value of -1 disables this feature. Useful for saving interim models during long training sessions.
            device=[0,1,2,3], # Specifies the computational device(s) for training: a single GPU (device=0), multiple GPUs (device=0,1), CPU (device=cpu), or MPS for Apple silicon (device=mps).
            workers=6, # Number of worker threads for data loading (per RANK if Multi-GPU training). Influences the speed of data preprocessing and feeding into the model, especially useful in multi-GPU setups.
            project="/home/sur06423/project/Version_5_exp/YOLO_v8/v8_n", # Name of the project directory where training outputs are saved. Allows for organized storage of different experiments.
            pretrained = True, # Determines whether to start training from a pretrained model. Can be a boolean value or a string path to a specific model from which to load weights. Enhances training efficiency and model performance.
            optimizer = "auto", # Choice of optimizer for training. Options include SGD, Adam, AdamW, NAdam, RAdam, RMSProp etc., or auto for automatic selection based on model configuration. Affects convergence speed and stability.
            verbose = True, # Enables verbose output during training, providing detailed logs and progress updates. Useful for debugging and closely monitoring the training process.
            seed = 0, # Sets the random seed for training, ensuring reproducibility of results across runs with the same configurations.
            amp = True, # Enables Automatic Mixed Precision (AMP) training, reducing memory usage and possibly speeding up training with minimal impact on accuracy.
            profile =True, # Enables profiling of ONNX and TensorRT speeds during training, useful for optimizing model deployment.
            val=True, # Enables validation during training, allowing for periodic evaluation of model performance on a separate dataset.
            plots = True, #Generates and saves plots of training and validation metrics, as well as prediction examples, providing visual insights into model performance and learning progression.                
)

New https://pypi.org/project/ultralytics/8.3.13 available 😃 Update with 'pip install -U ultralytics'
Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:1 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:2 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:3 (NVIDIA RTX A6000, 48677MiB)


engine/trainer: task=detect, mode=train, model=yolov8n.pt, data=/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml, epochs=100, time=None, patience=10, batch=128, imgsz=640, save=True, save_period=1, cache=False, device=[0, 1, 2, 3], workers=6, project=/home/sur06423/project/Version_5_exp/YOLO_v8/v8_n, name=train, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=True, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_json=False, save_hybrid=False, conf=None, iou=0.7, max_det=300, half=False, dnn=False, plots=True, source=None, vid_stride=1, stream_buffer=False, visualize=False, augment=False, agnostic_nms=False, classes=None, retina_masks=False, embed=None, show=False, save_frames=False, save_txt=False, save_conf=False, save_crop=False, show_labels=True, show_conf=True, sho

train: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/train/labels.cache... 4485 images, 430 backgrounds, 0 corrupt: 100%|██████████| 4485/4485 [00:00<?, ?it/s]
val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/valid/labels.cache... 1274 images, 124 backgrounds, 0 corrupt: 100%|██████████| 1274/1274 [00:00<?, ?it/s]


Plotting labels to /home/sur06423/project/Version_5_exp/YOLO_v8/v8_n/train/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.000714, momentum=0.9) with parameter groups 57 weight(decay=0.0), 64 weight(decay=0.001), 63 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 24 dataloader workers
Logging results to /home/sur06423/project/Version_5_exp/YOLO_v8/v8_n/train
Starting training for 100 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      1/100      4.67G       1.16        2.8      1.093          5        640: 100%|██████████| 36/36 [00:09<00:00,  3.89it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.87it/s]


                   all       1274       7377      0.999     0.0581      0.113     0.0938

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      2/100      4.52G      1.011       1.37     0.9965         17        640: 100%|██████████| 36/36 [00:06<00:00,  5.27it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.88it/s]


                   all       1274       7377      0.988     0.0711      0.244       0.17

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      3/100      4.43G     0.9713      1.195     0.9883         37        640: 100%|██████████| 36/36 [00:06<00:00,  5.25it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.90it/s]


                   all       1274       7377      0.821      0.232      0.323      0.196

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      4/100      4.46G     0.9484      1.075     0.9918          8        640: 100%|██████████| 36/36 [00:06<00:00,  5.35it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.09it/s]


                   all       1274       7377      0.674      0.283      0.329      0.196

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      5/100      4.66G     0.9412     0.9695       0.98          6        640: 100%|██████████| 36/36 [00:06<00:00,  5.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.09it/s]


                   all       1274       7377      0.716      0.365       0.42      0.258

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      6/100      4.45G     0.9124     0.8386     0.9611         26        640: 100%|██████████| 36/36 [00:06<00:00,  5.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.09it/s]


                   all       1274       7377      0.741      0.429      0.472      0.289

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      7/100      4.53G     0.8974     0.7772     0.9547          3        640: 100%|██████████| 36/36 [00:07<00:00,  4.89it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.38it/s]


                   all       1274       7377      0.758       0.44      0.503      0.299

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      8/100      4.46G      0.908     0.7515     0.9472         24        640: 100%|██████████| 36/36 [00:06<00:00,  5.40it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.36it/s]


                   all       1274       7377      0.771      0.441      0.488      0.293

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      9/100      4.48G     0.8732     0.6942     0.9577          1        640: 100%|██████████| 36/36 [00:06<00:00,  5.39it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.42it/s]


                   all       1274       7377      0.784      0.472      0.531       0.32

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     10/100      4.43G     0.8768      0.669     0.9425          8        640: 100%|██████████| 36/36 [00:06<00:00,  5.35it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.41it/s]


                   all       1274       7377      0.759      0.489      0.533      0.325

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     11/100      4.44G     0.8878     0.6608     0.9348         14        640: 100%|██████████| 36/36 [00:06<00:00,  5.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.47it/s]


                   all       1274       7377      0.759      0.478      0.526      0.319

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     12/100      4.46G     0.8907     0.7125     0.9251          1        640: 100%|██████████| 36/36 [00:06<00:00,  5.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.52it/s]


                   all       1274       7377      0.749      0.477      0.533      0.324

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     13/100      4.52G     0.8633      0.626     0.9326         11        640: 100%|██████████| 36/36 [00:06<00:00,  5.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.36it/s]


                   all       1274       7377      0.779      0.482      0.545      0.331

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     14/100      4.43G     0.8657     0.6439     0.9321         14        640: 100%|██████████| 36/36 [00:06<00:00,  5.36it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.51it/s]


                   all       1274       7377      0.801       0.49      0.556      0.341

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     15/100      4.46G     0.8473     0.6024     0.9279         24        640: 100%|██████████| 36/36 [00:06<00:00,  5.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.45it/s]


                   all       1274       7377      0.771      0.489      0.547      0.333

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     16/100       4.5G     0.8308     0.5851     0.9278          4        640: 100%|██████████| 36/36 [00:06<00:00,  5.37it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.52it/s]


                   all       1274       7377      0.814      0.493      0.558      0.347

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     17/100      4.51G     0.8485     0.6036     0.9384         15        640: 100%|██████████| 36/36 [00:06<00:00,  5.35it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.54it/s]


                   all       1274       7377      0.794      0.474      0.552      0.345

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     18/100      4.57G     0.8354     0.6048     0.9457          6        640: 100%|██████████| 36/36 [00:06<00:00,  5.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.52it/s]


                   all       1274       7377      0.781      0.503      0.567      0.347

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     19/100      4.75G      0.851     0.5903     0.9262         12        640: 100%|██████████| 36/36 [00:06<00:00,  5.41it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.43it/s]


                   all       1274       7377       0.77      0.515      0.572      0.348

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     20/100      4.69G     0.8172     0.5653     0.9258         23        640: 100%|██████████| 36/36 [00:06<00:00,  5.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.48it/s]


                   all       1274       7377      0.793      0.507      0.573      0.358

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     21/100      4.51G     0.8171      0.552      0.923          4        640: 100%|██████████| 36/36 [00:06<00:00,  5.35it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.48it/s]


                   all       1274       7377      0.818      0.523       0.59      0.363

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     22/100       4.7G     0.8434     0.5893      0.927          3        640: 100%|██████████| 36/36 [00:06<00:00,  5.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.47it/s]


                   all       1274       7377      0.821      0.513      0.585      0.362

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     23/100      4.23G     0.7957     0.5393     0.9188         27        640: 100%|██████████| 36/36 [00:06<00:00,  5.25it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.49it/s]


                   all       1274       7377        0.8      0.525      0.586      0.361

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     24/100      4.46G     0.7917     0.5294     0.9137          3        640: 100%|██████████| 36/36 [00:06<00:00,  5.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.45it/s]


                   all       1274       7377      0.836      0.514      0.593      0.374

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     25/100      4.47G      0.783      0.547     0.9148         20        640: 100%|██████████| 36/36 [00:06<00:00,  5.34it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.36it/s]


                   all       1274       7377      0.829      0.516      0.595       0.37

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     26/100       4.2G     0.7815     0.5196     0.9269         18        640: 100%|██████████| 36/36 [00:06<00:00,  5.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.00it/s]


                   all       1274       7377      0.831      0.529      0.598       0.37

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     27/100      4.51G     0.8174     0.5466     0.9147         11        640: 100%|██████████| 36/36 [00:06<00:00,  5.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.50it/s]


                   all       1274       7377      0.822      0.537      0.601      0.373

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     28/100      4.64G     0.7782     0.5206       0.91         20        640: 100%|██████████| 36/36 [00:06<00:00,  5.24it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.98it/s]


                   all       1274       7377      0.832      0.534      0.606      0.377

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     29/100       4.7G     0.7576     0.5014      0.901          7        640: 100%|██████████| 36/36 [00:06<00:00,  5.34it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.54it/s]


                   all       1274       7377      0.828      0.528      0.605      0.383

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     30/100      4.45G     0.7916       0.52     0.9151         11        640: 100%|██████████| 36/36 [00:06<00:00,  5.21it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.42it/s]


                   all       1274       7377      0.842      0.519      0.594      0.371

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     31/100      4.69G     0.7658     0.5119     0.9095         19        640: 100%|██████████| 36/36 [00:06<00:00,  5.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.50it/s]


                   all       1274       7377      0.844      0.535      0.609      0.377

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     32/100      4.69G     0.7664     0.5022     0.9021         10        640: 100%|██████████| 36/36 [00:06<00:00,  5.36it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.50it/s]


                   all       1274       7377      0.827      0.543      0.613      0.381

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     33/100      4.84G      0.754     0.4926     0.9049         13        640: 100%|██████████| 36/36 [00:06<00:00,  5.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.47it/s]


                   all       1274       7377      0.833      0.539      0.605      0.381

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     34/100      4.49G     0.7663     0.4925     0.9089         11        640: 100%|██████████| 36/36 [00:06<00:00,  5.25it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.36it/s]


                   all       1274       7377      0.833      0.545      0.613      0.384

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     35/100       4.7G     0.7663      0.494     0.9119         21        640: 100%|██████████| 36/36 [00:06<00:00,  5.34it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.47it/s]


                   all       1274       7377      0.842      0.551      0.623      0.389

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     36/100      4.68G       0.75     0.4934     0.9042          9        640: 100%|██████████| 36/36 [00:07<00:00,  5.13it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.48it/s]


                   all       1274       7377       0.83      0.551      0.619      0.394

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     37/100      4.42G     0.7587     0.4892     0.9066         10        640: 100%|██████████| 36/36 [00:06<00:00,  5.40it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.53it/s]


                   all       1274       7377      0.842      0.552      0.624      0.397

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     38/100      4.41G     0.7281     0.4862     0.8878          0        640: 100%|██████████| 36/36 [00:06<00:00,  5.37it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.44it/s]


                   all       1274       7377      0.845      0.553      0.628      0.397

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     39/100      4.66G     0.7407     0.4727     0.8993         21        640: 100%|██████████| 36/36 [00:06<00:00,  5.45it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.52it/s]


                   all       1274       7377      0.859      0.555       0.63      0.399

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     40/100      4.65G     0.7406     0.4799     0.8993         15        640: 100%|██████████| 36/36 [00:06<00:00,  5.61it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.49it/s]


                   all       1274       7377      0.835      0.556      0.628      0.396

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     41/100      4.41G     0.7251     0.4689     0.8971         10        640: 100%|██████████| 36/36 [00:06<00:00,  5.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.49it/s]


                   all       1274       7377      0.841      0.551       0.62      0.388

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     42/100      4.49G     0.7379     0.4652      0.897         24        640: 100%|██████████| 36/36 [00:06<00:00,  5.21it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.04it/s]


                   all       1274       7377      0.849      0.547      0.632      0.401

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     43/100      4.47G     0.7287     0.4696     0.9094          4        640: 100%|██████████| 36/36 [00:07<00:00,  5.11it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.43it/s]


                   all       1274       7377      0.835      0.558      0.625      0.397

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     44/100      4.93G     0.7459     0.4646     0.8865         26        640: 100%|██████████| 36/36 [00:06<00:00,  5.27it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.36it/s]


                   all       1274       7377      0.836      0.565      0.639      0.403

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     45/100      4.47G     0.7197     0.4561      0.895          8        640: 100%|██████████| 36/36 [00:06<00:00,  5.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.43it/s]


                   all       1274       7377      0.837      0.558      0.631      0.398

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     46/100      4.69G     0.7112      0.449     0.8974          8        640: 100%|██████████| 36/36 [00:06<00:00,  5.29it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.39it/s]


                   all       1274       7377       0.82      0.554      0.633      0.403

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     47/100      4.37G     0.7103     0.4524       0.89         15        640: 100%|██████████| 36/36 [00:06<00:00,  5.15it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.43it/s]


                   all       1274       7377      0.823      0.567      0.629      0.399

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     48/100      4.42G       0.73     0.4492     0.8893         16        640: 100%|██████████| 36/36 [00:06<00:00,  5.36it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.47it/s]


                   all       1274       7377      0.849      0.558      0.631      0.398

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     49/100      4.57G     0.7541     0.5127     0.8759          7        640: 100%|██████████| 36/36 [00:06<00:00,  5.37it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  5.00it/s]


                   all       1274       7377      0.847      0.553      0.624      0.395

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     50/100      4.44G      0.733     0.4513     0.8927         17        640: 100%|██████████| 36/36 [00:06<00:00,  5.37it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.54it/s]


                   all       1274       7377      0.825      0.581      0.647      0.411

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     51/100      4.51G     0.7248     0.4446     0.9022         10        640: 100%|██████████| 36/36 [00:06<00:00,  5.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.52it/s]


                   all       1274       7377      0.803      0.584      0.642      0.408

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     52/100      4.49G     0.7011     0.4359     0.8923          5        640: 100%|██████████| 36/36 [00:06<00:00,  5.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.49it/s]


                   all       1274       7377      0.837      0.567      0.638      0.404

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     53/100      4.37G     0.7057     0.4375      0.894          8        640: 100%|██████████| 36/36 [00:06<00:00,  5.36it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.47it/s]


                   all       1274       7377      0.838      0.567       0.64      0.407

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     54/100      4.42G     0.6832     0.4309     0.8866         14        640: 100%|██████████| 36/36 [00:06<00:00,  5.36it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.39it/s]


                   all       1274       7377      0.844       0.57       0.64      0.406

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     55/100       4.5G     0.7137     0.4513     0.8931         21        640: 100%|██████████| 36/36 [00:06<00:00,  5.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.51it/s]


                   all       1274       7377      0.833      0.586      0.651      0.414

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     56/100      4.66G     0.6891     0.4469     0.8919          5        640: 100%|██████████| 36/36 [00:06<00:00,  5.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.56it/s]


                   all       1274       7377      0.841      0.579      0.653      0.415

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     57/100      4.52G     0.7015     0.4387     0.8821         13        640: 100%|██████████| 36/36 [00:06<00:00,  5.17it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.62it/s]


                   all       1274       7377      0.841      0.582      0.655      0.419

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     58/100      4.47G     0.6952     0.4299     0.8929         21        640: 100%|██████████| 36/36 [00:07<00:00,  5.04it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.57it/s]


                   all       1274       7377      0.836      0.583      0.653      0.415

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     59/100      4.44G     0.6955     0.4261     0.8849         12        640: 100%|██████████| 36/36 [00:06<00:00,  5.29it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.53it/s]


                   all       1274       7377      0.843       0.58      0.657      0.418

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     60/100      4.61G     0.6922     0.4253     0.8858         18        640: 100%|██████████| 36/36 [00:06<00:00,  5.48it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.53it/s]


                   all       1274       7377      0.867      0.582       0.66      0.422

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     61/100      4.45G     0.7008     0.4375     0.8848         39        640: 100%|██████████| 36/36 [00:06<00:00,  5.41it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.54it/s]


                   all       1274       7377      0.863      0.573      0.649      0.419

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     62/100      4.66G     0.6759     0.4253     0.8827         33        640: 100%|██████████| 36/36 [00:06<00:00,  5.27it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.51it/s]


                   all       1274       7377      0.866      0.571      0.649      0.419

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     63/100      4.42G     0.7036     0.4331     0.8838         13        640: 100%|██████████| 36/36 [00:06<00:00,  5.34it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.54it/s]


                   all       1274       7377       0.86      0.578      0.651      0.418

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     64/100      4.47G      0.708      0.429     0.8835          6        640: 100%|██████████| 36/36 [00:06<00:00,  5.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.50it/s]


                   all       1274       7377       0.84      0.585      0.655       0.42

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     65/100      4.49G     0.7172     0.4408     0.8852          3        640: 100%|██████████| 36/36 [00:06<00:00,  5.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.52it/s]


                   all       1274       7377      0.839      0.584      0.662      0.427

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     66/100      4.64G     0.6863     0.4149     0.8886         10        640: 100%|██████████| 36/36 [00:06<00:00,  5.39it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.92it/s]


                   all       1274       7377      0.844      0.595       0.66      0.424

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     67/100       4.7G     0.7103       0.43     0.8875         19        640: 100%|██████████| 36/36 [00:07<00:00,  5.13it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.53it/s]


                   all       1274       7377      0.841      0.596      0.663      0.424

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     68/100      4.47G     0.6792      0.412     0.8774         19        640: 100%|██████████| 36/36 [00:06<00:00,  5.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.66it/s]


                   all       1274       7377      0.855      0.584      0.661      0.425

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     69/100      4.41G     0.7016     0.4244     0.8862         16        640: 100%|██████████| 36/36 [00:06<00:00,  5.51it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.48it/s]


                   all       1274       7377      0.865       0.57      0.655       0.42

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     70/100      4.42G     0.6914     0.4167     0.8814         24        640: 100%|██████████| 36/36 [00:06<00:00,  5.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.43it/s]


                   all       1274       7377      0.825      0.592      0.655      0.422

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     71/100       4.7G     0.6749     0.4019     0.8814         17        640: 100%|██████████| 36/36 [00:06<00:00,  5.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.41it/s]


                   all       1274       7377      0.838      0.597      0.666      0.428

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     72/100      4.66G     0.6794      0.404      0.879         12        640: 100%|██████████| 36/36 [00:06<00:00,  5.20it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.56it/s]


                   all       1274       7377      0.841      0.594      0.663      0.428

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     73/100      4.64G     0.6745     0.4035      0.884         13        640: 100%|██████████| 36/36 [00:06<00:00,  5.36it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.54it/s]


                   all       1274       7377      0.848      0.596      0.668      0.432

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     74/100      4.35G     0.6675     0.4008     0.8873         11        640: 100%|██████████| 36/36 [00:06<00:00,  5.49it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.51it/s]


                   all       1274       7377      0.854      0.594      0.668      0.429

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     75/100      4.69G     0.6694     0.4015     0.8758          6        640: 100%|██████████| 36/36 [00:06<00:00,  5.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.51it/s]


                   all       1274       7377      0.847      0.594      0.661      0.426

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     76/100      4.49G      0.684     0.4047     0.8737          8        640: 100%|██████████| 36/36 [00:06<00:00,  5.38it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.53it/s]


                   all       1274       7377      0.854       0.59      0.668      0.429

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     77/100      4.46G     0.6715     0.4024     0.8756         22        640: 100%|██████████| 36/36 [00:06<00:00,  5.24it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.67it/s]


                   all       1274       7377      0.834      0.598      0.665      0.429

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     78/100      4.69G     0.6788     0.3929     0.8957          4        640: 100%|██████████| 36/36 [00:06<00:00,  5.18it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.58it/s]


                   all       1274       7377      0.866      0.594      0.673      0.436

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     79/100      4.64G     0.6725     0.3985     0.8831         13        640: 100%|██████████| 36/36 [00:06<00:00,  5.37it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.53it/s]


                   all       1274       7377       0.85      0.602      0.672      0.434

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     80/100      4.69G     0.6584     0.3978     0.8764         36        640: 100%|██████████| 36/36 [00:06<00:00,  5.19it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.58it/s]


                   all       1274       7377       0.87      0.595      0.675      0.437

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     81/100      4.86G     0.6639     0.3961     0.8784          7        640: 100%|██████████| 36/36 [00:06<00:00,  5.44it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.39it/s]


                   all       1274       7377      0.841        0.6      0.668      0.432

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     82/100      4.56G     0.6627     0.3983     0.8759          9        640: 100%|██████████| 36/36 [00:06<00:00,  5.49it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.53it/s]


                   all       1274       7377      0.845      0.604      0.674      0.438

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     83/100      4.71G     0.6725     0.3997     0.8767          8        640: 100%|██████████| 36/36 [00:06<00:00,  5.42it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.51it/s]


                   all       1274       7377      0.857      0.601      0.671      0.434

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     84/100      4.45G     0.6708     0.4009     0.8778          8        640: 100%|██████████| 36/36 [00:06<00:00,  5.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.37it/s]


                   all       1274       7377      0.855      0.601      0.669      0.434

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     85/100      4.52G     0.6647     0.3964     0.8816         11        640: 100%|██████████| 36/36 [00:06<00:00,  5.34it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.44it/s]


                   all       1274       7377      0.843      0.607      0.672      0.434

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     86/100      4.41G     0.6847     0.4706     0.9334          3        640: 100%|██████████| 36/36 [00:06<00:00,  5.36it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.80it/s]


                   all       1274       7377      0.844      0.603       0.67      0.435

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     87/100      4.37G     0.6624     0.3903     0.8814         14        640: 100%|██████████| 36/36 [00:07<00:00,  5.10it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.59it/s]


                   all       1274       7377      0.859      0.601      0.671      0.438

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     88/100      4.49G     0.6595     0.3864     0.8734         16        640: 100%|██████████| 36/36 [00:06<00:00,  5.24it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.44it/s]


                   all       1274       7377      0.859      0.606      0.678      0.441

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     89/100      4.67G     0.6215     0.3679      0.876         14        640: 100%|██████████| 36/36 [00:06<00:00,  5.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.51it/s]


                   all       1274       7377      0.856      0.606      0.675      0.439

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     90/100      4.47G     0.6429     0.3843     0.8843          3        640: 100%|██████████| 36/36 [00:06<00:00,  5.47it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.21it/s]


                   all       1274       7377      0.861      0.604      0.673      0.437
Closing dataloader mosaic

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     91/100      4.49G     0.7132     0.3889     0.8639         13        640: 100%|██████████| 36/36 [00:07<00:00,  4.69it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.54it/s]


                   all       1274       7377      0.869      0.602      0.677      0.434

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     92/100      4.47G     0.7071     0.3868      0.866         13        640: 100%|██████████| 36/36 [00:06<00:00,  5.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.98it/s]


                   all       1274       7377      0.853      0.606      0.675      0.437

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     93/100      4.41G     0.7088     0.3866       0.86          7        640: 100%|██████████| 36/36 [00:06<00:00,  5.35it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.54it/s]


                   all       1274       7377      0.864      0.598      0.674      0.435

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     94/100       4.5G     0.7111      0.392     0.8611          6        640: 100%|██████████| 36/36 [00:06<00:00,  5.23it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.44it/s]


                   all       1274       7377      0.864      0.599      0.676       0.44

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     95/100      4.53G     0.7171     0.3919     0.8598         31        640: 100%|██████████| 36/36 [00:06<00:00,  5.53it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.52it/s]


                   all       1274       7377      0.868      0.599      0.676       0.44

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     96/100      4.42G     0.7195     0.3983     0.8671         13        640: 100%|██████████| 36/36 [00:06<00:00,  5.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.47it/s]


                   all       1274       7377      0.866      0.602       0.68      0.443

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     97/100      4.42G     0.7101     0.3874     0.8608         13        640: 100%|██████████| 36/36 [00:06<00:00,  5.43it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.49it/s]


                   all       1274       7377      0.874      0.598      0.681      0.442

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     98/100      4.41G     0.7129     0.3887      0.854          3        640: 100%|██████████| 36/36 [00:06<00:00,  5.46it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.52it/s]


                   all       1274       7377      0.877        0.6      0.683      0.442

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     99/100      4.37G     0.6997     0.3795     0.8584         40        640: 100%|██████████| 36/36 [00:06<00:00,  5.52it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.55it/s]


                   all       1274       7377      0.871      0.602       0.68      0.441

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    100/100      4.47G      0.693     0.3766     0.8572          6        640: 100%|██████████| 36/36 [00:06<00:00,  5.58it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:03<00:00,  5.53it/s]


                   all       1274       7377      0.869      0.602      0.682      0.443

100 epochs completed in 0.335 hours.
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_v8/v8_n/train/weights/last.pt, 6.3MB
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_v8/v8_n/train/weights/best.pt, 6.3MB

Validating /home/sur06423/project/Version_5_exp/YOLO_v8/v8_n/train/weights/best.pt...
Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:1 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:2 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:3 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 168 layers, 3,006,428 parameters, 0 gradients, 8.1 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:07<00:00,  2.84it/s]


                   all       1274       7377      0.867      0.602      0.681      0.442
            Pedestrian        333        600      0.797       0.61      0.685       0.39
         Traffic_Signs        309        434      0.876       0.52      0.598       0.34
               Vehicle       1034       2820      0.906      0.855      0.907       0.72
         traffic_light        660       3523      0.892      0.422      0.535      0.319
Speed: 0.1ms preprocess, 1.0ms inference, 0.0ms loss, 0.6ms postprocess per image
Results saved to /home/sur06423/project/Version_5_exp/YOLO_v8/v8_n/train


# Results on Validation set:
100 epochs completed in 0.335 hours.
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_v8/v8_n/train/weights/last.pt, 6.3MB
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_v8/v8_n/train/weights/best.pt, 6.3MB

Validating /home/sur06423/project/Version_5_exp/YOLO_v8/v8_n/train/weights/best.pt...
Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:1 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:2 (NVIDIA RTX A6000, 48677MiB)
                                                           CUDA:3 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 168 layers, 3,006,428 parameters, 0 gradients, 8.1 GFLOPs
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:07<00:00,  2.84it/s]
                   all       1274       7377      0.867      0.602      0.681      0.442
            Pedestrian        333        600      0.797       0.61      0.685       0.39
         Traffic_Signs        309        434      0.876       0.52      0.598       0.34
               Vehicle       1034       2820      0.906      0.855      0.907       0.72
         traffic_light        660       3523      0.892      0.422      0.535      0.319
Speed: 0.1ms preprocess, 1.0ms inference, 0.0ms loss, 0.6ms postprocess per image
Results saved to /home/sur06423/project/Version_5_exp/YOLO_v8/v8_n/train

# Test the results on Test dataset
- The above results was on the validation set of the Carla_labeling-v5 dataset (See the Image counts)
- Now we need to calculate the metrics on the test set of the Carla_labeling-v5 dataset containing only 641 images.
- Below is the script to do so:

In [6]:
from ultralytics import YOLO

# Load YOLOv8n Last epoch weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v8/v8_n/train/weights/last.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.25, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)


Model summary (fused): 168 layers, 3,006,428 parameters, 0 gradients, 8.1 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:08<00:00,  1.42s/it]


                   all        641       3701      0.857      0.595      0.753      0.534
            Pedestrian        182        332      0.807      0.617      0.749      0.467
         Traffic_Signs        168        230      0.832      0.496      0.695      0.447
               Vehicle        528       1377      0.921      0.855      0.918      0.785
         traffic_light        338       1762      0.868       0.41      0.652      0.436
Speed: 0.4ms preprocess, 1.1ms inference, 0.0ms loss, 2.5ms postprocess per image
Saving runs/detect/val/predictions.json...
Results saved to runs/detect/val


# Results on Last checpoint (100th epoch):
Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 168 layers, 3,006,428 parameters, 0 gradients, 8.1 GFLOPs
val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:08<00:00,  1.42s/it]
                   all        641       3701      0.857      0.595      0.753      0.534
            Pedestrian        182        332      0.807      0.617      0.749      0.467
         Traffic_Signs        168        230      0.832      0.496      0.695      0.447
               Vehicle        528       1377      0.921      0.855      0.918      0.785
         traffic_light        338       1762      0.868       0.41      0.652      0.436
Speed: 0.4ms preprocess, 1.1ms inference, 0.0ms loss, 2.5ms postprocess per image
Saving runs/detect/val/predictions.json...
Results saved to runs/detect/val

# Results on Test Set using Best Checkpoint:

In [7]:
from ultralytics import YOLO

# Load YOLOv8n Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v8/v8_n/train/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.25, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

Ultralytics YOLOv8.2.93 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)


Model summary (fused): 168 layers, 3,006,428 parameters, 0 gradients, 8.1 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:08<00:00,  1.35s/it]


                   all        641       3701      0.857      0.595      0.753      0.534
            Pedestrian        182        332      0.807      0.617      0.749      0.467
         Traffic_Signs        168        230      0.832      0.496      0.695      0.447
               Vehicle        528       1377      0.921      0.855      0.918      0.785
         traffic_light        338       1762      0.868       0.41      0.652      0.436
Speed: 0.3ms preprocess, 1.0ms inference, 0.0ms loss, 2.1ms postprocess per image
Saving runs/detect/val2/predictions.json...
Results saved to runs/detect/val2


In [15]:
print(checkpoint.keys())

dict_keys(['date', 'version', 'license', 'docs', 'epoch', 'best_fitness', 'model', 'ema', 'updates', 'optimizer', 'train_args', 'train_metrics', 'train_results'])


# Taking the Optimal value from the F1-Confidence curve for Optimal Confidence
- Optimal Confidence YOLOv8n - 0.247
- IoU threshold : 0.5
- Results save d at: runs/detect/val3

In [1]:
from ultralytics import YOLO

# Load YOLOv8n Last epoch weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v8/v8_n/train/weights/last.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.247, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.5, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 168 layers, 3,006,428 parameters, 0 gradients, 8.1 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:07<00:00,  1.22s/it]


                   all        641       3701      0.862      0.595      0.754      0.534
            Pedestrian        182        332      0.817       0.62      0.753      0.468
         Traffic_Signs        168        230      0.832      0.496      0.695      0.447
               Vehicle        528       1377       0.93      0.853      0.918      0.785
         traffic_light        338       1762       0.87       0.41      0.653      0.437
Speed: 1.9ms preprocess, 1.8ms inference, 0.0ms loss, 3.3ms postprocess per image
Saving runs/detect/val3/predictions.json...
Results saved to runs/detect/val3


# Best + Taking the Optimal value from the F1-Confidence curve for Optimal Confidence
- Optimal Confidence YOLOv8n - 0.247
- IoU threshold : 0.5
- Results save d at: runs/detect/val4

In [1]:
from ultralytics import YOLO

# Load YOLOv8n Last epoch weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v8/v8_n/train/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.247, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.5, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 168 layers, 3,006,428 parameters, 0 gradients, 8.1 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:07<00:00,  1.22s/it]


                   all        641       3701      0.862      0.595      0.754      0.534
            Pedestrian        182        332      0.817       0.62      0.753      0.468
         Traffic_Signs        168        230      0.832      0.496      0.695      0.447
               Vehicle        528       1377       0.93      0.853      0.918      0.785
         traffic_light        338       1762       0.87       0.41      0.653      0.437
Speed: 1.9ms preprocess, 1.3ms inference, 0.0ms loss, 2.2ms postprocess per image
Saving runs/detect/val4/predictions.json...
Results saved to runs/detect/val4


# Best Epoch 100 + Optimal Confidence + IoU 0.6
- Optimal Confidence YOLOv8n - 0.247
- IoU threshold : 0.6
- Results save d at: runs/detect/val5

In [1]:
from ultralytics import YOLO

# Load YOLOv8n Last epoch weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_v8/v8_n/train/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.247, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 168 layers, 3,006,428 parameters, 0 gradients, 8.1 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:07<00:00,  1.26s/it]


                   all        641       3701      0.856      0.595      0.753      0.534
            Pedestrian        182        332      0.805       0.62       0.75      0.467
         Traffic_Signs        168        230      0.832      0.496      0.695      0.447
               Vehicle        528       1377      0.919      0.855      0.917      0.785
         traffic_light        338       1762      0.867       0.41      0.652      0.436
Speed: 1.9ms preprocess, 1.3ms inference, 0.0ms loss, 2.6ms postprocess per image
Saving runs/detect/val5/predictions.json...
Results saved to runs/detect/val5


# Seperate evaluation on the training set

In [1]:
from ultralytics import YOLO

# Load your model
model = YOLO('/home/sur06423/project/Version_5_exp/YOLO_v8/v8_n/train/weights/best.pt')  # replace with your model path

# Validate on custom dataset
results = model.val(data='/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml',
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.25, # Sets the optimal confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="train", 
)  # Ensure your YAML points to your desired dataset

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
Model summary (fused): 168 layers, 3,006,428 parameters, 0 gradients, 8.1 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/train/labels.cache... 4485 images, 430 backgrounds, 0 corrupt: 100%|██████████| 4485/4485 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 36/36 [00:22<00:00,  1.57it/s]


                   all       4485      26788      0.899      0.625      0.786      0.583
            Pedestrian       1229       2227      0.848      0.665      0.797       0.55
         Traffic_Signs       1093       1459      0.901      0.542      0.742      0.498
               Vehicle       3636      10031      0.933      0.877      0.931      0.808
         traffic_light       2373      13071      0.914      0.414      0.675      0.475
Speed: 0.4ms preprocess, 1.1ms inference, 0.0ms loss, 0.8ms postprocess per image
Saving runs/detect/val6/predictions.json...
Results saved to runs/detect/val6


# This notebook is complete.
- All the cell are executed and saved for later use.
- Results are saved in the corresponding directories.